dockers/docker-orchagent/orchagent.sh

```diff
@@ -17,6 +17,22 @@ ORCHAGENT_ARGS="-d /var/log/swss "
 # Set orchagent pop batch size to 8192
 ORCHAGENT_ARGS+="-b 8192 "
 
+# Check if there is an "asic_id field" in the DEVICE_METADATA in configDB.
+#"DEVICE_METADATA": {
+#    "localhost": {
+#        ....
+#        "asic_id": "0",
+#    }
+#},
+# ID field could be integers just to denote the asic instance like 0,1,2...
+# OR could be PCI device ID's which will be strings like "03:00.0"
+# depending on what the SAI/SDK expects.
+asic_id=`sonic-cfggen -d -v DEVICE_METADATA.localhost.asic_id`
+if [ -n "$asic_id" ]
+then
+    ORCHAGENT_ARGS+="-i $asic_id "
```

Comment:
>$asic_id [](start = 24, length = 8)

There is security risk that attacker tamper the asic_id in Redis and run any command.

Reply:
good call.

Reply:
i put this as a todo list.

Reply:
Sure Guohan, will take it up as a follow up action. But wanted to add one more point to this.

Currently we pass the instance ID as integers 0,1,2. It will be changed in future to pass the PCI device ID's when we update the new SAI will accepts PCI device IDs. The config DB will be populated with PCI device ID's in this field "DEVICE_METADATA.localhost.asic_id" and will be passed on when starting orchagent process.

This should be safe ? if we pass a wrong PCI device id ( may be by an attacker) ... the SAI create switch will fail and syncd won't come up...